test/connection.test.js

```diff
@@ -1547,4 +1547,12 @@ describe('connections:', function() {
       // currently cannot write test for continueOnError or errors in general.
     });
   });
+  describe('processConnectionOptions', function() {
+    it('should not throw an error when attempting to mutate unmutable options object gh-13335', async function() {
+      const m = new mongoose.Mongoose();
+      const opts = Object.preventExtensions({});
+      const conn = await m.connect('mongodb://localhost:27017/db?retryWrites=true&w=majority&readPreference=secondaryPreferred', opts);
```

Comment:
Maybe on github runner the /mongo database server cant use 27017 as a port and claims a different port, thus breaking the test.

Maybe use getUri from /test/common.js and retrieve the url to the database server.

Maybe create Mongo Client Adapter and pass it to mongoose and check if opts is not the same object as passed to the adapters connect function. It should then even work without starting a mongo database server.